src/test/storage/constraints_test.cpp

```diff
@@ -0,0 +1,376 @@
+#include <memory>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "concurrency/transaction_manager.hpp"
+#include "expression/expression_functional.hpp"
+#include "expression/pqp_column_expression.hpp"
+#include "operators/delete.hpp"
+#include "operators/get_table.hpp"
+#include "operators/insert.hpp"
+#include "operators/projection.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/update.hpp"
+#include "storage/constraints/unique_checker.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+class ConstraintsTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    column_definitions.emplace_back("column0", DataType::Int, false);
+    column_definitions.emplace_back("column1", DataType::Int, true);
+    column_definitions.emplace_back("column2", DataType::Int, false);
+
+    auto table_temp = std::make_shared<Table>(column_definitions, TableType::Data, 3, UseMvcc::Yes);
+    auto& manager = StorageManager::get();
+    manager.add_table("table_temp", table_temp);
+
+    table_temp->append({1, 1, 3});
+    table_temp->append({2, 1, 2});
+    table_temp->append({3, 2, 0});
+
+    auto gt = std::make_shared<GetTable>("table_temp");
+    gt->execute();
+
+    auto table = std::make_shared<Table>(column_definitions, TableType::Data, 3, UseMvcc::Yes);
+    table->add_unique_constraint({ColumnID{0}});
+    manager.add_table("table", table);
+    auto table_insert = std::make_shared<Insert>("table", gt);
+    auto table_context = TransactionManager::get().new_transaction_context();
+    table_insert->set_transaction_context(table_context);
+    table_insert->execute();
+    table_context->commit();
+
+    nullable_column_definitions.emplace_back("column0", DataType::Int, true);
+    nullable_column_definitions.emplace_back("column1", DataType::Int, false);
+    nullable_column_definitions.emplace_back("column2", DataType::Int, false);
```

Comment:
I'm a little irritated by the naming. Both column definition structs `column_definitions`& `nullable_column_definitions` define two non-nullable and a one nullable column. 
The naming implies more fundamental differences. Or am I misunderstanding something?